src/GameServer/Systems/Player/ProgressManager.cs

```diff
@@ -42,10 +45,71 @@ public async Task<bool> UnlockTransPoint(int sceneId, int pointId)
             var update = Builders<ProgressManager>.Update.Set(w => w.UnlockedScenePoints, UnlockedScenePoints);
             await DatabaseManager.UpdateProgressAsync(filter, update);
             //await Owner.Inventory.AddItemByIdAsync(201, 5, ActionReason.UnlockPointReward);
-            // await Owner.Inventory.AddItemByIdAsync(102, isStatue ? 50 : 10, ActionReason.UnlockPointReward);
+            //await Owner.Inventory.AddItemByIdAsync(102, isStatue ? 50 : 10, ActionReason.UnlockPointReward);
             await Owner.SendPacketAsync(new PacketScenePointUnlockNotify((uint)sceneId, (uint)pointId));
             return true;
         }
+        public async Task<bool> UnlockTransPoint(List<int> sceneId, Dictionary<int, HashSet<uint>> pointIds)
+        {
+            foreach (int a in sceneId)
+            {
+                if (!UnlockedScenePoints.ContainsKey(a))
+                {
+                    
+                    if (pointIds[a] != null)
+                    {
+                        UnlockedScenePoints.Add(a, new HashSet<int>());
+                        foreach (int i in pointIds[a])
+                        {
+                            UnlockedScenePoints[a].Add(i);
+                            List<uint> list = new List<uint>();
+                            list.AddRange(pointIds[a]);
+                            await Owner.SendPacketAsync(new PacketScenePointUnlockNotify(a, list));
```

Comment:
with the above changes you can pass pointIds[a] directly
```suggestion
                            await Owner.SendPacketAsync(new PacketScenePointUnlockNotify(a, pointIds[a]));
```